zerocopy-derive/src/lib.rs

```diff
@@ -238,8 +210,112 @@ fn derive_known_layout_inner(ast: &DeriveInput, _top_level: Trait) -> Result<Tok
+                //  - its fields are enumerated in the order they appear,
+                //  - the presence of `repr_align` and `repr_packed` are correctly accounted for.
+                //
+                // We respect all three of these preconditions here. This
+                // expansion is only used if `is_repr_c_struct`, we enumerate
+                // the fields in order, and we extract the values of `align(N)`
+                // and `packed(N)`.
+                const LAYOUT: ::zerocopy::DstLayout = {
+                    use ::zerocopy::util::macro_util::core_reexport::num::NonZeroUsize;
+                    use ::zerocopy::{DstLayout, KnownLayout};
+
+                    let repr_align = #repr_align;
+                    let repr_packed = #repr_packed;
+
+                    DstLayout::new_zst(repr_align)
+                        #(.extend(DstLayout::for_type::<#leading_fields_tys>(), repr_packed))*
+                        .extend(<#trailing_field_ty as KnownLayout>::LAYOUT, repr_packed)
+                        .pad_to_align()
+                };
+
+                #methods
+            )
+        };
+
+        let outer_extras = {
+            let ident = &ast.ident;
+            let vis = &ast.vis;
+            let params = &ast.generics.params;
+            let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+
+            let predicates = if let Some(where_clause) = where_clause {
+                where_clause.predicates.clone()
+            } else {
+                Default::default()
+            };
+
+            let methods = make_methods(&parse_quote! {
+                <#trailing_field_ty as ::zerocopy::KnownLayout>::MaybeUninit
+            });
+
+            quote! {
+                // SAFETY: This has the same layout as the derive target type,
+                // except that it admits uninit bytes. This is ensured by using the
+                // same repr as the target type, and by using field types which have
+                // the same layout as the target type's fields, except that they
+                // admit uninit bytes.
+                #repr
+                #[doc(hidden)]
+                #vis struct __ZerocopyKnownLayoutMaybeUninit<#params> (
+                    #(::zerocopy::util::macro_util::core_reexport::mem::MaybeUninit<#leading_fields_tys>,)*
+                    <#trailing_field_ty as ::zerocopy::KnownLayout>::MaybeUninit
+                )
+                where
+                    #trailing_field_ty: ::zerocopy::KnownLayout,
+                    #predicates;
+
+                // SAFETY: We largely defer to the `KnownLayout` implementation on
+                // the derive target type (both by using the same tokens, and by
+                // deferring to impl via type-level indirection). This is sound,
+                // since  `__ZerocopyKnownLayoutMaybeUninit` is guaranteed to be
+                // identical to that of the derive target type, except that
```

Comment:
```suggestion
                // since  `__ZerocopyKnownLayoutMaybeUninit` is guaranteed to have
                // the same layout as the derive target type, except that
```

Reply:
[Done](https://github.com/google/zerocopy/compare/bb140d07cfd265378c2a38132b7d3b0ba8dbd81a..b573d832ca02f2b735359c87dd92de151e7d8b33#diff-977c29e6cf079bd7307f811df24f99633df603962785c583145eb6250b0e12daL292-R293)